virtualizarr/manifests/array.py

```diff
@@ -170,7 +189,7 @@ def __eq__(  # type: ignore[override]
         if self.shape != other.shape:
             raise NotImplementedError("Unsure how to handle broadcasting like this")
 
-        if not utils.metadata_identical(self.metadata, other.metadata):
+        if not self.metadata.to_dict() == other.metadata.to_dict():
```

Comment:
This comment explains why we are now able to simplify this: https://github.com/zarr-developers/zarr-python/pull/2930#issuecomment-3077641365